### PR TITLE
error messages: disable text-based inline code styling

### PR DESCRIPTION
**Pull request by Octachron**
This PR disables the text-based styling of inline code while keeping the bold styling when the terminal supports it.

This avoids potential confusion with OCaml's `"` within error messages. 
I was half-tempted to replace the use of `"` for quoting inline code by typographic quote `“…”` but since quotes are not really needed, it is probably better to avoid introducing unnecessary unicode characters.

This PR also amends the error message 

> Existential types are not allowed in let ... and ... bindings

that was too ambiguous without quotes:

> Existential types are not allowed in grouped (let ... and ...) bindings

Similarly, expect tests keep a visible textual quote style to help catch missing `inline_code` tags.

See also #12210 and @dbuenzli 's comment.

### CODE REVIEW

File: utils/misc.ml [hunk 1]
```diff
@@ -863,7 +863,7 @@ module Style = struct
       error = no_markup [Bold; FG Red];
       loc = no_markup [Bold];
       hint = no_markup [Bold; FG Blue];
-      inline_code= { ansi=[Bold]; text_open = {|"|}; text_close = {|"|} }
+      inline_code= no_markup [Bold]
```
Comment (ghost) [inline utils/misc.ml:866]: After this change, the `text_close` and `text_open` fields of `tag_style` are never set, so the code of `mark_open_tag` and `mark_close_tag` further down in this file could be simplified and these fields removed completely.

Comment (Octachron) [reply utils/misc.ml:866]: These fields are set in `testsuite/tools/expect.ml` to make the `inline code` tag visible in the testsuite without resorting to ANSI escape code.

Comment (ghost) [reply utils/misc.ml:866]: Damn, I had not looked at the changes in `testsuite/` yet. Sorry for the noise.

**Review by goldfirere — APPROVED** [review]
I like this change. We had made a similar change within Jane Street, but I think this one is better.

**Pull request merged**